### ISSUE TITLE
MinGW's linker does not create a reloc section for DYNAMICBASE executables

### ISSUE THREAD

**Issue by thestinger** [rust-lang/rust#16581]
This issue exists to track the necessity of a workaround for this bug in the compiler.

**Comment by steveklabnik** [comment]
Triage: unsure if there's been any change.

**Comment by mati865** [comment]
@rustbot modify labels: -O-windows +O-windows-gnu

Fixed in Binutils 2.34: https://sourceware.org/bugzilla/show_bug.cgi?id=17321
There is reliable workaround for older versions: https://github.com/TheRyuu/FFmpeg/commit/91b668acd6decec0a6f8d20bf56e2644f96adcb9
LLD was never affected.

**Comment by mati865** [comment]
AFAIK this is fixed in Binutils shipped on recent nightly builds.

**Comment by mati865** [comment]
Yeah, Rust now ships Binutils 2.42: https://github.com/rust-lang/rust/pull/119229
